src/meta/Intersection.ts

```diff
@@ -0,0 +1,167 @@
+import global from '@dojo/shim/global';
+import WeakMap from '@dojo/shim/WeakMap';
+import { Base } from './Base';
+
+interface IntersectionDetail {
+	entries: WeakMap<Element, IntersectionObserverEntry>;
+	intersectionObserver?: IntersectionObserver; // attached observer
+	options: IntersectionGetOptions;
+	root: string;
+	rootMargin: string | undefined;
+	thresholds: number[]; // thresholds the observe should be attached with
+}
+
+export interface IntersectionGetOptions {
+	root?: string;
+	rootMargin?: string;
+	threshold?: number;
```

Comment:
Do we need `threshold` and `thresholds`? If we need a convenience of passing a single threshold, can we not just type it `number | number[]`?

Reply:
My preference would be to go with thresholds and make it always `number[]`